doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,350 @@
+
+Both Python and OpenSSL methods are automatically handled for the user, whereas using custom commands requires more configuration steps.
+
+Checking the public key
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You can check that the bootloader image is correctly compiled with the custom signing key by comparing its auto-generated public key against a manual public key dump using OpenSSL.
+To do this you can use ``diff``, running the following command from a terminal:
+
+.. code-block:: console
+
+   diff build/zephyr/nrf/subsys/bootloader/generated/public.pem <(openssl ec -in priv.pem -pubout)
+
+If there is no file diff output, then the private key was successfully included with the bootloader image.
+
+.. _ug_bootloader_adding_immutable_b0_custom_signing:
+
+Custom signing commands
+-----------------------
+
+If you want complete control over the key handling of a project, the |NSIB| supports the use of a custom signing command.
+Using a custom signing command, instead of calling upon a specific command and public key for signing the firmware images, removes the use of a private key from the build system.
+This is useful when the private keys are stored, managed, or otherwise processed through a *hardware security module* (`HSM`_) or in-house tool.
+
+To use a custom signing command with this bootloader, set the following options in the application's ``prj.conf`` file, in an associated Kconfig fragment, or using the command line:
+
+.. tabs::
+
+   .. group-tab:: Kconfig / prj.conf
+
+      .. code-block:: console
+
+         CONFIG_SECURE_BOOT=y
+         CONFIG_SB_SIGNING_CUSTOM=y
+         CONFIG_SB_SIGNING_PUBLIC_KEY="/path/to/pub.pem"
+         CONFIG_SB_SIGNING_COMMAND="my_command"
+
+   .. group-tab:: Command line
+
+      .. code-block:: console
+
+         west build -b nrf52840dk_nrf5840 zephyr/samples/hello_world -- \
+         -DCONFIG_SECURE_BOOT=y \
+         -DCONFIG_SB_SIGNING_CUSTOM=y \
+         -DCONFIG_SB_SIGNING_PUBLIC_KEY=\"/path/to/pub.pem\" \
+         -DCONFIG_SB_SIGNING_COMMAND=\"my_command\"
+
+      Escaped quotations avoid malformed-string warnings from Kconfig.
+
+.. note::
+
+   The public key string must be an absolute path to the location of the public key file.
+   Environment variables (like ``$HOME``,``$PWD``, or ``$USER``) and the ``~`` character on Unix systems are not expanded when setting an absolute path from a ``prj.conf`` file or Kconfig fragment, but are expanded correctly in key file strings on the command line.
+
+See :option:`CONFIG_SB_SIGNING_COMMAND` for specifics about what a signing command must do to be usable.
+The command string can include its own arguments like a typical terminal command, with arguments sent to the command by the build system being appended to the end:
+
+.. parsed-literal::
+   :class: highlight
+
+   my_command *[options]* *<args ...>* *<build_system_args ..>*
+
+Arguments sent to the command by the build system in this way are noted in :option:`CONFIG_SB_SIGNING_COMMAND`.
```

Comment:
Does this happen automatically (passive) or is it something you must or can edit (active)?

Passive voice here makes it unclear what I'm actually in control of as the user.

Reply:
Reworded, please see latest changes.